src/NSubstitute/Extensions/ReceivedExtensions.cs

```diff
@@ -155,5 +157,32 @@ public override bool Equals(object obj)
 
             public override int GetHashCode() { return 0; }
         }
+
+        private class RangeQuantity : Quantity
+        {
+            private readonly int minInclusive;
+            private readonly int maxInclusive;
+            public RangeQuantity(int minInclusive, int maxInclusive) {
+                if (minInclusive < 0) {
+                    throw new ArgumentOutOfRangeException(nameof(minInclusive),
+                        $"{nameof(minInclusive)} must be >= 0, but was {minInclusive}.");
+                }
+                if (maxInclusive <= minInclusive) {
```

Comment:
Should we allow ranges like [2;2]? [According to the wiki](https://en.wikipedia.org/wiki/Interval_(mathematics)) such ranges are still valid from math point of view:

> ... Note that (a, a), [a, a), and (a, a] each represents the empty set, whereas [a, a] denotes the set {a}.

So while it's weird to have ranges including single value, formally it's still fine to support them.

Reply:
This is a tricky one. I _think_ it might be better for usability to detect and throw here so it makes it obvious they are specifying a single value (particularly if it isn't obvious due to variable use, say `Received(Quantity.Within(a, b))`).  If people want a singleton range they can say `Received(2)`, which I think would always be preferable to `Received(Quantity.Within(2,2))`. 😄 

If you feel strongly about it I am happy to change it, as I'm really not sure, but my initial feeling is the stricter input format may help avoid some accidental errors. 🤷‍♂

Reply:
I don't have _strong_ opinion on that. My consideration is that there is nothing wrong with [2;2] - it's just weird to do that. I think API should prevent from absurd things only.

It might also enable clients to write parametrized tests, so that could use the same check to assert exact quantity (using various `TestCase`).

But anyway, I'm shifting decision making to you, so whatever you decide will be right and wrong in the same time 😊

Reply:
I started to implement this, and have decided that, seeing as I'm going to be wrong anyway, I'll leave it as it is, for the very logical reason that the default description of "between 2 and 2 (inclusive) calls" looks really strange and I don't want to put more logic in the description to simplify it to "exactly 2". 🤣